### PR TITLE
Keep a static representation of strings for low indexes [0...999].

### PR DESCRIPTION
**Pull request by baldersheim**
This will avoid blowing up the heap with many duplicate strings.

@bratseth or @lesters or @bjorncs PR

### CODE REVIEW

File: vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java [hunk 1]
```diff
@@ -13,6 +13,21 @@
  * @author bratseth
  */
 public abstract class TensorAddress implements Comparable<TensorAddress> {
+    private static final String [] SMALL_INDEXES = createSmallIndexesAsStrings(1000);
```
Comment (bjorncs) [inline vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java:16]: Space between `String` and `[]` looks very much like a C programmer coding Java ;)

**Review by bjorncs — APPROVED** [review]

File: vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java (hunk 1 above)
Comment (baldersheim) [reply vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java:16]: :)

Comment (kkraune) [reply vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java:16]: priceless

**Review by bratseth — COMMENTED** [review]
👍

**Pull request merged**